tests/kernel/mem_protect/obj_validation/src/main.c

```diff
@@ -112,10 +121,109 @@ void test_generic_object(void)
 }
 
+/**
+ * @brief Test requestor thread will implicitly be assigned permission on the
+ * dynamically allocated object
+ *
+ * @details
+ * - Create kernel object semaphore, dynamically allocate it from the calling
+ *   thread's resource pool.
+ * - Check that object's address is in bounds of that memory pool.
+ * - Then check the requestor thread will implicitly be assigned permission on
+ *   the allocated object by using semaphore API k_sem_init()
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_object_alloc()
+ */
+void test_kobj_assign_perms_on_alloc_obj(void)
+{
+	test_dyn_sem0 = k_object_alloc(K_OBJ_SEM);
+	zassert_not_null(test_dyn_sem0, "Cannot allocate sem k_object");
+
+	struct k_thread *thread = _current;
+
+	intptr_t sem_addr = (intptr_t)&test_dyn_sem0;
+	intptr_t start_addr = (intptr_t)&thread->resource_pool;
+	size_t size_heap = CONFIG_HEAP_MEM_POOL_SIZE;
+
+	/* check that kernel object may be dynamically allocated,
+	 * drawing memory from the calling thread's memory pool
+	 */
+#if defined(CONFIG_X86)
+	zassert_true(
+		(sem_addr > start_addr) && (sem_addr <= (start_addr + size_heap)),
+		"semaphore not in bound of thread's memory pool");
+#elif defined(CONFIG_ARM)
+	zassert_true(
+		(sem_addr > start_addr) && (sem_addr <= (start_addr + size_heap)),
+		"semaphore not in bound of thread's memory pool");
+#elif defined(CONFIG_ARC)
+	zassert_true(
+		(sem_addr < start_addr) && (sem_addr > (start_addr - size_heap)),
+		"semaphore not in bound of thread's memory pool");
+#else
+#error "Not implemented for this architecture"
+	zassert_unreachable("Write to control register fault");
+#endif
+
+	k_sem_init(test_dyn_sem0, 1, 1);
+	zassert_equal(k_sem_count_get(test_dyn_sem0), 1,
+			"sem count not equal to the init count");
+}
+
+/* Helper function inits dynamic kernel object and then revokes access */
+static void dyn_object_release(void *p1, void *p2, void *p3)
+{
+	struct z_object *ko;
+
+	test_dyn_mutex = k_object_alloc(K_OBJ_MUTEX);
+	zassert_not_null(&test_dyn_mutex, "Can't allocate dynamical kobj");
+
+	struct k_thread *thread = _current;
+
+	ko = z_object_find(test_dyn_mutex);
+	zassert_true(ko, NULL);
+
+	k_object_access_revoke(test_dyn_mutex, thread);
+}
+
+/**
+ * @brief Test dynamically allocated kernel object release memory
+ *
+ * @details Dynamically allocated kernel objects whose access is controlled by
+ * the permission system will use object permission as a reference count.
+ * If no threads have access to an object, the object's memory released.
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_object_alloc()
+ */
+void test_no_ref_dyn_kobj_release_mem(void)
+{
+	k_thread_create(&user_thr0, user_stack_thr0,
+			 K_THREAD_STACK_SIZEOF(user_stack_thr0),
+			 dyn_object_release,
+			 NULL, NULL, NULL,
+			 K_INHERIT_PERMS, 0, K_NO_WAIT);
+
+	k_thread_join(&user_thr0, K_FOREVER);
+
+	struct z_object *ko = z_object_find(test_dyn_mutex);
+
+	ko = z_object_find(test_dyn_mutex);
+
+	zassert_false(ko, NULL);
```

Comment:
please don't pass NULL to zassert_ functions for the description. Use a string that describes what happened and why it is not good.

Reply:
Understood, I wrote code according to the existing zassert code that uses NULL and I thought it is acceptable.